orchestra/analytics/latency.py

```diff
@@ -128,3 +132,55 @@ def project_time_row_generator(project_information,
             [iteration.end_datetime for iteration in iterations])
         for start_datetime, iteration in zip(iteration_times[:-1], iterations):
             yield iteration.to_dict(start_datetime)
+
+
+def work_time_aggregate(df, dimensions):
+    """
+    Aggregates work time grouped by `dimensions`.
+
+    Args:
+        df (pandas.DataFrame):
+            A DataFrame with a schema described in `work_time_df`.
+        dimensions (list):
+            A list of column names to group on.
+
+    Returns:
+        df (pandas.DataFrame):
+            A DataFrame with aggregate information.
+    """
+    return df.groupby(dimensions, as_index=False).agg({'work_time': np.sum})
+
+
+def calendar_time_aggregate(df, dimensions):
+    """
+    Aggregates calendar time grouped by `dimensions`.
+
+    Args:
+        df (pandas.DataFrame):
+            A DataFrame with a schema described in `work_time_df`.
+        dimensions (list):
+            A list of column names to group on.
+
+    Returns:
+        df (pandas.DataFrame):
+            A DataFrame with aggregate information.
+    """
+    # When reporting the project time, we want to avoid double-counting two
+    # tasks that are running in parallel, so we calculate the difference
+    # between the farthest-apart start and end times for iterations.  For more
+    # granular groupings like assignments, because iterations are
+    # interspersed, computing the end time minus the start time would double-
+    # count calendar time, so we want to sum up their computed
+    # calendar_datetimes.
+    if dimensions == ['project_id']:
+        aggregated = df.groupby(dimensions, as_index=False).agg({
+            'start_datetime': np.min,
+            'end_datetime': np.max})
+        aggregated['calendar_time'] = (
+            aggregated['end_datetime'] - aggregated['start_datetime'])
+        aggregated.drop('start_datetime', axis=1, inplace=True)
+        aggregated.drop('end_datetime', axis=1, inplace=True)
+        return aggregated
```

Comment:
I think so, but vet this:
- For automatically assigned tasks, the calendar time is nearly 0, so the start time of iterations 0 and 1 are nearly identical.
- For non-automatically assigned tasks, the calendar time is non-zero, depending on when an expert picks up a task.    The calendar time of the task/project, however, should include this non-zero time, which is captured by its earlier `start_datetime`.